ale/util.py

```diff
@@ -99,3 +103,72 @@ def find_latest_metakernel(path, year):
     if not metakernel:
         raise Exception(f'No metakernels found in {path} for {year}.')
     return metakernel
+
+
+def generate_metakernel_from_isis(cube, mkpath=None):
+    isis_preferences = os.path.join(os.path.expanduser("~"), '.Isis', 'IsisPreferences')
+
+    if not os.path.isfile(isis_preferences):
+        isis_preferences = os.path.join(pysis.env.ISIS_ROOT, 'IsisPreferences')
+    try:
+        with open(isis_preferences) as f:
+            preftext = f.read().replace('EndGroup', 'End_Group')
+            pvlprefs = pvl.loads(preftext)
+    except Exception as e:
+        raise Exception(f'Failed to load IsisPreferences file [{isis_preferences}]: {e}')
+
+    # enforce key order
+    mk_paths = OrderedDict.fromkeys(
+        ['TargetPosition', 'InstrumentPosition',
+         'InstrumentPointing', 'Frame', 'TargetAttitudeShape',
+         'Instrument', 'InstrumentAddendum', 'LeapSecond',
+         'SpacecraftClock', 'Extra'])
+
+    # just work with full path
+    cube = os.path.abspath(cube)
+    cubelabel = pvl.load(cube)
+    kernel_group = cubelabel['IsisCube']['Kernels']
+
+    # Start loading. Stuff that might have multiple entries first
+    mk_paths['TargetPosition'] = kernel_group.get('TargetPosition', None)[1:]
+    mk_paths['InstrumentPosition'] = kernel_group.get('InstrumentPosition', None)[1:]
+    mk_paths['InstrumentPointing'] = kernel_group.get('InstrumentPointing', None)[1:]
+    mk_paths['TargetAttitudeShape'] = [kernel_group.get('TargetAttitudeShape', None)]
```

Comment:
There's a chance these won't have "Table" at the beginning, a more explicit check may be better. This is likely a corner case with old data though so I'm not sure.